client/my-sites/plans-features-main/index.tsx

```diff
@@ -820,7 +843,12 @@ const PlansFeaturesMain = ( {
 										coupon={ coupon }
 										currentSitePlanSlug={ sitePlanSlug }
 										generatedWPComSubdomain={ resolvedSubdomainName }
-										gridPlanForSpotlight={ gridPlanForSpotlight }
+										gridPlanForSpotlight={
+											gridPlansForFeaturesGrid.length >= 4 ||
+											hideLowerTierPlansExperimentAssignment?.variationName !== 'treatment'
+												? gridPlanForSpotlight
+												: undefined
+										}
```

Comment:
If control, pass `gridPlanForSpotlight`.
If treatment and `gridPlansForFeaturesGrid.length < 4`, pass `gridPlanForSpotlight`.
If treatment and `gridPlansForFeaturesGrid.length >= 4`, pass `undefined`.